File: clojure/scrabble-score/src/scrabble_score.clj
(ns scrabble-score)

(def score->letters
  {1 [\A \E \I \O \U \L \N \R \S \T]
   2 [\D \G]
   3 [\B \C \M \P]
   4 [\F \H \V \W \Y]
   5 [\K]
   8 [\J \X]
   10 [\Q \Z]})

(def letter->score 
  (into {} 
        (for [[score letters] score->letters
              letter letters] 
          [letter score])))

(defn score-word
  "Returns the scrabble score of a word."
  [word]
  (reduce + (map letter->score (clojure.string/upper-case word))))

File: python/scrabble-score/scrabble_score.py
"""Compute a word's Scrabble score"""

SCORE_BANDS = {
    'AEIOULNRST': 1, 'DG': 2, 'BCMP': 3, 
    'FHVWY': 4, 'K': 5, 'JX': 8, 'QZ': 10
}

LETTER_SCORES = {
    letter: score
    for letters, score in SCORE_BANDS.items()
    for letter in letters
}

def score(word):
    return sum(LETTER_SCORES[letter] for letter in word.upper())
